binding/web/tests/factory/selenium_test.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/python3
+
+import os
+import sys
+import threading
+import time
+from argparse import ArgumentParser
+from http.server import HTTPServer, SimpleHTTPRequestHandler
+
+from selenium import webdriver
+from selenium.webdriver.chrome.options import Options
+from selenium.webdriver.common.desired_capabilities import DesiredCapabilities
+
+
+def simple_http_server(host='localhost', port=4001, path='.'):
+    server = HTTPServer((host, port), SimpleHTTPRequestHandler)
+    thread = threading.Thread(target=server.serve_forever)
+    thread.deamon = True
+
+    cwd = os.getcwd()
+    base_url = f'http://{host}:{port}'
+
+    def start():
+        os.chdir(path)
+        thread.start()
+        print(f'starting server on port {server.server_port}')
+
+    def stop():
+        os.chdir(cwd)
+        server.shutdown()
+        server.socket.close()
+        print(f'stopping server on port {server.server_port}')
+
+    return start, stop, base_url
+
+
+def main():
+    parser = ArgumentParser()
+
+    parser.add_argument(
```

Comment:
do we need this arg? can't we just derive it ourselves based on the location of this script?